### PR TITLE
Create batting_and_pitching_splits

### PR DESCRIPTION
**Pull request by mwisnie5**
This module pulls split stats from bbref. New PR to pull in split stats from bbref. This supports both pitching and hitting split stats and I listed out the splits available in the comments.

### CODE REVIEW

File: pybaseball/batting_and_pitching_splits [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        ump - By Umpire
+
+    """
+    if season is None:
+        season='career'
+    season=str(season)
+    links=[]
+    for playerid in playerids:
+        for split in splits:
+            url=["https://widgets.sports-reference.com/wg.fcgi?css=1&site=br&url=%2Fplayers%2Fsplit.fcgi%3Fid%3D"+playerid+"%26year%3D"+season+"%26t%3D"+PitchOrBat+"&div=div_"+split]
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:131]: change this to an f-string and remove from brackets

File: pybaseball/batting_and_pitching_splits [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    """
+    if season is None:
+        season='career'
+    season=str(season)
+    links=[]
+    for playerid in playerids:
+        for split in splits:
+            url=["https://widgets.sports-reference.com/wg.fcgi?css=1&site=br&url=%2Fplayers%2Fsplit.fcgi%3Fid%3D"+playerid+"%26year%3D"+season+"%26t%3D"+PitchOrBat+"&div=div_"+split]            
+            links+=url
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:132]: `links.append(url)`

File: pybaseball/batting_and_pitching_splits [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+def download_url(url):
+    """
+    Creates a local file with the contents of each table
+    """
+    resp=requests.get(url)
+    title="".join(x for x in url if x.isalpha())+"html"
+    
+    with open(title, "wb") as fh:
+        fh.write(resp.content)
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:14]: why are we writing this to a file? parse this locally

File: pybaseball/batting_and_pitching_splits [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    
+def download_tables(table_urls):
+    threads=min(30, len(table_urls))
+    
+    with concurrent.futures.ThreadPoolExecutor(max_workers=max(1,threads)) as executor:
+        executor.map(download_url, table_urls)
+        
+
+def read_split_files(urls):
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:23]: annotate

File: pybaseball/batting_and_pitching_splits [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    """
+    reads the split files that were created and outputs a dataframe of stats by split. 
+    """
+    html=[]
+    data=[]
+    for i in range(len(urls)):
+        with open("".join(x for x in urls[i] if x.isalpha())+"html") as f:
+            html=f.read()
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:31]: don't run this thru file IO

File: pybaseball/batting_and_pitching_splits [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            data.append([ele for ele in cols])
+    data = pd.DataFrame(data)
+    data = data.rename(columns=data.iloc[0])
+    data = data.reindex(data.index.drop(0))
+    data = data.set_index(['Player ID', 'Split Type', 'Split'])
+    data = data.drop(index=['Split'], 
+                            level=2)
+    data = data.apply(pd.to_numeric, errors='coerce')
+    data = data.fillna(0.0)
+    data = data.iloc[:,:-3]
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:58]: this line needs a comment -- what's this doing?

File: pybaseball/batting_and_pitching_splits [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    data['1B']=data['H']-data['2B']-data['3B']-data['HR'] #populates the 1B column based on the other columns
+    return data
+
+def find_split_urls(playerids, splits, PitchOrBat='b', season=None):
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:62]: Our normal format would be `pitch_or_bat`.
Also, please add type annotations

File: pybaseball/batting_and_pitching_splits [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        dr - Days of Rest
+        hitlo - Hit Location
+        traj - Hit Trajectory
+        oppon - opponent splits
+        stad - game conditions  
+        site - ballpark
+        catch - By Catcher
+        ump - By Umpire
+
+    """
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:124]: I would reduce the docstring to more limited usage stuff and move the full documentation to its own file (see our standard documentation format)

File: pybaseball/batting_and_pitching_splits [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        for split in splits:
+            url=["https://widgets.sports-reference.com/wg.fcgi?css=1&site=br&url=%2Fplayers%2Fsplit.fcgi%3Fid%3D"+playerid+"%26year%3D"+season+"%26t%3D"+PitchOrBat+"&div=div_"+split]            
+            links+=url
+    download_tables(links) #gets remaining tables
+    data=read_split_files(links) #reads and combines tables into a single dataframe
+    return data
+
+#EXAMPLE USAGE
+#SplitMultiIndex=find_split_urls(['rizzoan01','troutmi01'], ['total', 'power', 'plato'], PitchOrBat='b')
+#print(SplitMultiIndex)
```
Comment (schorrm) [inline pybaseball/batting_and_pitching_splits:139]: example to lowercase, and document this using the standard format with the rest

File: pybaseball/batting_and_pitching_splits (hunk 7 above)
Comment (schorrm) [reply pybaseball/batting_and_pitching_splits:62]: I would consider a boolean flag here?

**Review by schorrm — CHANGES_REQUESTED** [review]

**Comment by schorrm** [comment]
@mwisnie5 -- this looks really good -- can you please take a look at my comments?

**Comment by mwisnie5** [comment]
> @mwisnie5 -- this looks really good -- can you please take a look at my comments?

Hey Moshe. Just saw these comments. Will take a look and make the changes needed.

File: pybaseball/batting_and_pitching_splits (hunk 6 above)
Comment (mwisnie5) [reply pybaseball/batting_and_pitching_splits]: removed that line. It was no longer needed.

File: pybaseball/batting_and_pitching_splits (hunk 8 above)
Comment (mwisnie5) [reply pybaseball/batting_and_pitching_splits]: I reduced the docstring, but haven't created the documentation file yet. Will work on that in the coming days.

**Comment by schorrm** [comment]
Sorry, we I cleaned up the history and this will need a pull and rebase etc. Very sorry.

**Comment by mwisnie5** [comment]
> Sorry, I cleaned up the history and this will need a pull and rebase etc. Very sorry.

Sorry. I'm new to github. What do you mean by this? Is this me submitting this as a new pull request and cancelling the existing one?

**Comment by schorrm** [comment]
That might work?

**Comment by mwisnie5** [comment]
Cancelled PR and will be adding an improved PR that only requires one call to bbref

**Pull request closed without merging**